core/types/src/kafka.rs

```diff
@@ -7,19 +11,35 @@ pub enum KafkaTopics {
 impl AsRef<str> for KafkaTopics {
     fn as_ref(&self) -> &str {
         match self {
-            KafkaTopics::ChainEvents => "chain_events",
-            KafkaTopics::MpPrices => "mp_prices",
+            KafkaTopics::ChainEvents => "chain-events",
+            KafkaTopics::MpPrices => "mp-prices",
         }
     }
 }
 
-
 impl From<&str> for KafkaTopics {
     fn from(value: &str) -> Self {
         match value {
-            "chain_events" => KafkaTopics::ChainEvents,
-            "mp_prices" => KafkaTopics::MpPrices,
-            _ => unimplemented!(),
+            "chain-events" => KafkaTopics::ChainEvents,
+            "mp-prices" => KafkaTopics::MpPrices,
+            _ => unimplemented!("Kafka topic is not valid"),
         }
     }
 }
+
+#[derive(Serialize, Deserialize, Clone, Debug)]
+pub struct ChainBlock {
+    pub chain_id: u64,
+    pub block_number: u64,
+    pub block_timestamp: Option<u64>,
+    pub events: Vec<ChainEvent>,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug)]
+pub struct ChainEvent {
+    pub parsed_event: Option<Value>,
```

Comment:
I think there is either no need to hold parsed or row event, my design was to only hold row event struct and parse it where nesessarry